### ISSUE TITLE
Feature request: Sleep/suspend controls 

### ISSUE THREAD

**Issue by Dinth** [bkbilly/lnxlink#23]
**Describe the problem**

**OS Version**

Linux Mint 21
**LNXLink version** 

9c7e3fbf7c7b6267a13c52fab06253277b94a855

I would like to request a feature of enabling/disabling sleep on the machine 
(it can be done by the following commands:
```
sudo systemctl mask sleep.target suspend.target hibernate.target hybrid-sleep.target
sudo systemctl unmask sleep.target suspend.target hibernate.target hybrid-sleep.target
```
or alternatively by making lnxlink service inhibit sleep, simple example: https://github.com/bulletmark/sleep-inhibitor
This can be presented as a switch in Home Assistant.
On the top of that, it would be nice to have Sleep/Suspend commands which would work in a same way as Shutdown command right now.

**Comment by bkbilly** [comment]
For making the the PC Sleep/Suspend we can use this command like [this](https://askubuntu.com/a/1795/361714):
```bash
systemctl suspend
```

As for the enabling/disabling the automatic hibernation, I don't fully agree with the method of masking the systemd processes. We will have to use a more universal way of doing that like [this](https://askubuntu.com/a/1324131/361714):
```bash
xdotool key 0x00
```

**Comment by bkbilly** [comment]
I am not sure how to get the sleeping state, but the online/offline can be checked by checking if one of the services is unavailable, just like the example on Readme: https://github.com/bkbilly/lnxlink/#combine-with-wake-on-lan-to-control-your-pc-with-one-switch

**Comment by bkbilly** [comment]
@Dinth From your newest issue #26 it seems that it works.
Please confirm to close this Feature Request.

**Comment by Dinth** [comment]
Hard to say, if its in the "idle" module then it doesnt work for me, i needed to disable the module and I dont have any sleep controls exposed in HA.

**Comment by bkbilly** [comment]
For the sleep controls you need to add the module `keep_alive` on your configuration if it is not already there.
It will create a new switch on Home Assistant that you can toggle. It sends a null key using the `xdotool` command.

As for the idle module I'll leave the issue #26 open until I can find a way to work on Linux Mint.

**Comment by Dinth** [comment]
Cool. Would you mind leaving this open for a while longer? I seem to have some problems with the new version, probably due to the upgrade process (the previous version was in /opt, the new one installed in user folder). The daemon seems to working fine now, but its not reporting properly to HA. I will test this feature as soon as i manage to fix my problem.

**Comment by bkbilly** [comment]
Let me know if I can help. 
You can close this anytime you feel it is fixed.

**Comment by Dinth** [comment]
OK, i think i know what was the issue i have been experiencing, will create another ticket for that.
keep_alive seems to work for me fine :)

**Comment by Dinth** [comment]
Hmm i just installed it in headless mode and im getting something like that (i am not sure, but i believe it may be related to keep_alive):
```
● lnxlink.service - LNXLink
     Loaded: loaded (/etc/systemd/system/lnxlink.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2022-11-06 11:41:24 GMT; 6s ago
   Main PID: 7658 (lnxlink)
      Tasks: 1 (limit: 38212)
     Memory: 14.5M
        CPU: 59ms
     CGroup: /system.slice/lnxlink.service
             └─7658 /usr/bin/python3 /usr/local/bin/lnxlink -c /etc/lnxlink.yaml

Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]:     from pynput import keyboard
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]:   File "/usr/local/lib/python3.10/dist-packages/pynput/__init__.py", line 40, in <module>
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]:     from . import keyboard
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]:   File "/usr/local/lib/python3.10/dist-packages/pynput/keyboard/__init__.py", line 31, in <module>
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]:     backend = backend(__name__)
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]:   File "/usr/local/lib/python3.10/dist-packages/pynput/_util/__init__.py", line 76, in backend
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]:     raise ImportError('this platform is not supported: {}'.format(
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]: ImportError: this platform is not supported: ('failed to acquire X connection: Bad display name ""', DisplayNameError(''))
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]: Try one of the following resolutions:
Nov 06 11:41:30 dinth-desktop-linux lnxlink[7658]:  * Please make sure that you have an X server running, and that the DISPLAY environment variable is set correctly
```

**Comment by bkbilly** [comment]
The headless installation doesn't support some modules.
The error you get is from the send_keys module, but you have to try which ones are not supported.

**Comment by Iceman248** [comment]
Regarding headless installation, how does that compare to regular one? With what this app does, I don't see why there would be such a difference.

**Comment by Dinth** [comment]
The main difference is that the service works even if the user who installed it is not logged in. Potentially non-headless installation will work fine for people who had computers used by just one person who is always logged in (and they dont need monitoring before they log in). 
Another difference is that a couple of modules dont work with headless installation (weirdly, im pretty sure they used to work before when lnxlink was installed in /opt, or at least did not prevent lnxlink from starting though.)

**Comment by bkbilly** [comment]
The headless installation is for servers and it's not meant to be installed on desktops.
The main difference is that servers don't have Graphical Interface and that's why some modules don't work.

As for the old way of installation, they still had problem which you can find from the closed issues I had.

**Comment by Iceman248** [comment]
Okay, so I plan to use on 2 different computers, both with monitors attached with only 1 user per computer. I should use regular install then? If so, what is the current proper method for doing so. I was trying to update a test machine and can't even find the directory where it installed. It used to be in /opt/lnxlink. I couldn't sent a notification test.

**Comment by bkbilly** [comment]
I was supposed to write a documentation about upgrading from the previous version which was using `/opt/lnxlink` directory, but I didn't get the chance to do so.

The current version is installed as a python package which is exposed with the command `lnxlink` which is located on your home directory: `~/.local/bin/lnxlink`.

The configuration and systemd service are automatically created the first time you run the app.
You can follow the installation instructions: https://github.com/bkbilly/lnxlink#installation
Make sure when asked, select to run as a user service.

**Comment by Iceman248** [comment]
Where is the directory at that has the modules/config.yaml? I actually saw the config.yaml in the root of my home directory, but not anything else.

The part of the install had always been troublesome for me, although it seems to work without doing it, is the first line.

`sudo apt install patchelf meson libdbus-glib-1-dev libglib2.0-dev libasound2-dev`

Pretty much all of the second part can't be found. I have tried with Arch based distros, EndeavourOS and Manjaro.

It stopped working for me recently, don't even know how to manually run as the run.py is in an unknown location to me now.
